voxel51/fiftyone
# Delete non-persistent datasets at shutdown

This is an alternate to #1742. 

This allows us to retain the feature of automatically deleting non-persistent datasets. It is done when the last session disconnects from the database as opposed to start-up.

### Test

- Open a new terminal session (Session # 1) and run `python`:

    ```py 
    import fiftyone as fo
    dataset = fo.Dataset('test')
    session = fo.launch_app(dataset)
    ```

- Keep Session # 1 running.

- Open another terminal session (Session # 2):

    `fiftyone datasets list  # non-persistent dataset ('test') is listed`

- Exit python in Session # 1.

- Back in Session # 2:

    `fiftyone datasets list  # non-persistent dataset ('test') should no longer be listed`

Should work for both a built-in and external Mongo database.

### Config

If users wish to disable this feature they can do so by adding `disable_automatic_dataset_cleanup` to their config or setting the ENV variable `FIFTYONE_DISABLE_AUTOMATIC_DATASET_CLEANUP`.
